components/Typography/src/MDCFontScaler.m

```diff
@@ -0,0 +1,324 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCFontScaler.h"
+
+#import <objc/runtime.h>
+
+#import "UIFont+MaterialScalable.h"
+#import "private/MDCFontTraits.h"
+#import "private/UIContentSizeCategory+Material.h"
+
+MDCTextStyle const MDCTextStyleHeadline1 = @"MDC.TextStyle.Headline1";
+MDCTextStyle const MDCTextStyleHeadline2 = @"MDC.TextStyle.Headline2";
+MDCTextStyle const MDCTextStyleHeadline3 = @"MDC.TextStyle.Headline3";
+MDCTextStyle const MDCTextStyleHeadline4 = @"MDC.TextStyle.Headline4";
+MDCTextStyle const MDCTextStyleHeadline5 = @"MDC.TextStyle.Headline5";
+MDCTextStyle const MDCTextStyleHeadline6 = @"MDC.TextStyle.Headline6";
+MDCTextStyle const MDCTextStyleSubtitle1 = @"MDC.TextStyle.Subtitle1";
+MDCTextStyle const MDCTextStyleSubtitle2 = @"MDC.TextStyle.Subtitle2";
+MDCTextStyle const MDCTextStyleBody1 = @"MDC.TextStyle.Body1";
+MDCTextStyle const MDCTextStyleBody2 = @"MDC.TextStyle.Body2";
+MDCTextStyle const MDCTextStyleButton = @"MDC.TextStyle.Button";
+MDCTextStyle const MDCTextStyleCaption = @"MDC.TextStyle.Caption";
+MDCTextStyle const MDCTextStyleOverline = @"MDC.TextStyle.Overline";
+
+@implementation MDCFontScaler {
+  NSDictionary<UIContentSizeCategory, NSNumber *> *_scalingCurve;
+  MDCTextStyle _textStyle;
+}
+
++ (instancetype)scalerForMaterialTextStyle:(MDCTextStyle)textStyle {
+  return [[MDCFontScaler alloc] initForMaterialTextStyle:textStyle];
+}
+
+- (instancetype)initForMaterialTextStyle:(MDCTextStyle)textStyle {
+  self = [super init];
+  if (self) {
+    _textStyle = [textStyle copy];
+
+    // NOTE: All scaling curves MUST include a full set of values for ALL UIContentSizeCategory
+    // values. This values must not decrease and the category size increases. To put it another
```

Comment:
Nit: I think you mean, "must not decrease **as** the **size category** increases.?